src/client/Client.cc

```diff
@@ -7999,9 +7999,21 @@ int Client::_readdir_cache_cb(dir_result_t *dirp, add_dirent_cb_t cb, void *p,
   while (true) {
     if (!dirp->inode->is_complete_and_ordered())
       return -EAGAIN;
+    
+    Dentry *dn = *pd;
+    int r = _getattr(dn->inode, caps, dirp->perms);
+    if (r < 0)
+      return r;
+    
     if (pd == dir->readdir_cache.end())
       break;
-    Dentry *dn = *pd;
+    
+    Dentry *dn_readdir = *pd;
+    vector<Dentry*>::iterator ret;
+    ret = std::find(dir->readdir_cache.begin(), dir->readdir_cache.end(), dn_readdir);
```

Comment:
this is too inefficient.   something like

int idx = pd - readdir_cache.begin();
getattr(...)
pd = readdir_cache+ idx;
if (pd < readdir_cache.end() && *pd == dn)
   ...

Reply:
This is a good skill, thank you
Updated!